Sources/TuistServer/Session/ServerSessionController.swift

```diff
@@ -57,22 +71,25 @@ public final class ServerSessionController: ServerSessionControlling {
 
     // MARK: - ServerSessionControlling
 
-    public func authenticate(serverURL: URL) async throws {
+    public func authenticate(
+        serverURL: URL,
+        deviceCodeType: DeviceCodeType,
+        onOpeningBrowser: @escaping (URL) -> Void,
+        onAuthWaitBegin: () -> Void
+    ) async throws {
         var components = URLComponents(url: serverURL, resolvingAgainstBaseURL: false)!
         let deviceCode = uniqueIDGenerator.uniqueID()
-        components.path = "/auth/cli/\(deviceCode)"
-        components.queryItems = nil
+        components.path = "/auth/device_codes/\(deviceCode)"
+        components.queryItems = [
+            URLQueryItem(name: "type", value: deviceCodeType.rawValue),
```

Comment:
I find the binding between the value the API expects and `rawValue` very implicit and prone to break. What about having `deviceCodeType.apiValue`?